default_warehouse_from_sale_team/models/procurement_group.py

```diff
@@ -0,0 +1,23 @@
+from odoo import api, models
+
+
+class ProcurementGroup(models.Model):
+    _inherit = "procurement.group"
+
+    @api.model
+    def _search_rule(self, route_ids, product_id, warehouse_id, domain):
+        """Grant access to provided warehouse if it's being denied by this module's record rule
+
+        If computing an inventory rule to apply but current user has no access to the required warehouse
+        because it's not configured in their sales teams, ensure access is granted as that's an automatic
+        flow that shouldn't be restricted by permissions.
+        """
+        rule_warehouse = self.env.ref("default_warehouse_from_sale_team.rule_default_warehouse_wh")
+        if (
+            warehouse_id
+            and not self.env.su
+            and warehouse_id.check_access_rights("read", raise_exception=False)
+            and self.env["ir.rule"]._get_failing(warehouse_id, mode="read") == rule_warehouse
+        ):
+            warehouse_id = warehouse_id.sudo()
```

Comment:
why not by default always use sudo as the old approach?

Reply:
@luisg123v @hugho-ad I think that using sudo for everything we could also solve this case: https://www.vauxoo.com/web#id=17904&action=1918&model=helpdesk.ticket&view_type=form&cids=1&menu_id=1431

Reply:
@hugho-ad I would like to avoid that approach, as we would be forcin sudo in all situations, independently of whether is required.

FYI the other issue commented by @desdelinux is not solved by adding sudo, because it's an access error on a different model (picking).